pandas/core/ops.py

```diff
@@ -754,8 +754,9 @@ def wrapper(self, other, axis=None):
         elif isinstance(other, pd.DataFrame):  # pragma: no cover
             return NotImplemented
         elif isinstance(other, (np.ndarray, pd.Index)):
-            if len(self) != len(other):
-                raise ValueError('Lengths must match to compare')
+            if not lib.isscalar(lib.item_from_zerodim(other)):
+                if len(self) != len(other):
```

Comment:
still to much nested if statements
maybe just try to convert a 0-dim ndarray to a scaler instead